### PR TITLE
Towards moar type (in)sanity

### LINKED ISSUE #128: Fix types in ArbitraryTuple
From #127: is anyone brave enough to put some sanity in the types of this class? Typescript supports [variadic tuple types](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html) ever since v4.0, but we are clearly not using them correctly, and so we have a lot of unsafe casting going around.

### PR DESCRIPTION
**Pull request by hugoferreira**
WIP... `Record`s in `FluentCheck` and derived classes are a mess, tuples in `ArbitraryTuple` are a mess. This aims to decrease the mess surface. Will close #128 or die trying.

### CODE REVIEW

**Comment by hugoferreira** [comment]
FWIW, there are still unsafe `any`'s and type casts going around, but I think this is a better version than what we already had (and my head is strobing xD)

**Comment by hugoferreira** [comment]
Can we merge this?

File: src/strategies/FluentStrategy.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import {Arbitrary, FluentPick, FluentRandomGenerator} from '../arbitraries'
-import {FluentConfig, FluentResult} from '../FluentCheck'
+import {FluentResult} from '../FluentCheck'
 import {StrategyArbitraries} from './FluentStrategyTypes'
 
+export type FluentConfig = { sampleSize?: number, shrinkSize?: number }
```
Comment (ruippeixotog) [inline src/strategies/FluentStrategy.ts:5]: Does eslint like this?

**Review by ruippeixotog — COMMENTED** [review]
LGTM overall, but I am not overly familiar with the `FluentCheck` class so I'd prefer @arestivo to take a second look.

Please make sure `eslint` doesn't issue any warnings before merging.

File: src/strategies/FluentStrategy.ts (hunk 1 above)
Comment (hugoferreira) [reply src/strategies/FluentStrategy.ts:5]: Yes, and I don't know why 😭

```text
➜ fluent-check git:(yet-another-type-madness) ✗ npm run lint

> fluent-check@1.0.0 lint
> eslint --ext '.js,.ts' src test

➜ fluent-check git:(yet-another-type-madness) ✗ 
```

**Pull request merged**